compiler/backend/presLangScript.sml

```diff
@@ -234,381 +234,341 @@ val exh_to_pres_exp_def = tDefine"exh_to_pres_exp"`
+  /\
+  (pres_to_structured (Plit lit) =
+      Item NONE "Plit" [lit_to_structured lit])
+  /\
+  (pres_to_structured (Pcon conF exps) =
+    let exps' = List (MAP pres_to_structured exps) in
+      Item NONE "Pcon" [conf_to_structured conF; exps'])
+  /\
+  (pres_to_structured (Pref exp) =
+      Item NONE "Pref" [pres_to_structured exp])
+  /\
+  (pres_to_structured (Ptannot exp t) =
+      Item NONE "Ptannot" [pres_to_structured exp; t_to_structured t])
+  /\
+  (pres_to_structured (Raise tra exp) =
+      Item (SOME tra) "Raise" [pres_to_structured exp])
+  /\
+  (pres_to_structured (Handle tra exp expsTup) =
+    let expsTup' = List (MAP (\(e1, e2) . Tuple [ pres_to_structured e1; pres_to_structured e2 ]) expsTup) in
+      Item (SOME tra) "Handle" [pres_to_structured exp; expsTup'])
+  /\
+  (pres_to_structured (Var_local tra varN) =
+      Item (SOME tra) "Var_local" [string_to_structured varN])
+  /\
+  (pres_to_structured (Var_global tra num) =
+      Item (SOME tra) "Var_global" [num_to_structured num])
+  /\
+  (pres_to_structured (Extend_global tra num) =
+      Item (SOME tra) "Extend_global" [num_to_structured num])
+  /\
+  (pres_to_structured (Lit tra lit) =
+      Item (SOME tra) "Lit" [lit_to_structured lit])
+  /\
+  (pres_to_structured (Con tra conF exps) =
+    let exps' = List (MAP pres_to_structured exps) in
+      Item (SOME tra) "Pcon" [conf_to_structured conF; exps'])
+  /\
+  (pres_to_structured (App tra op exps) =
+    let exps' = List (MAP pres_to_structured exps) in
+      Item (SOME tra) "App" [op_to_structured op; exps'])
+  /\
+  (pres_to_structured (Fun tra varN exp) =
+      Item (SOME tra) "Fun" [string_to_structured varN; pres_to_structured exp])
+  /\
+  (pres_to_structured (Log tra lop exp1 exp2) =
+      Item (SOME tra) "Log" [lop_to_structured lop; pres_to_structured exp1; pres_to_structured exp2])
+  /\
+  (pres_to_structured (If tra exp1 exp2 exp3) =
+      Item (SOME tra) "If" [pres_to_structured exp1; pres_to_structured exp2; pres_to_structured exp3])
+  /\
+  (pres_to_structured (Mat tra exp expsTup) =
+    let expsTup' = List (MAP (\(e1, e2) . Tuple [ pres_to_structured e1; pres_to_structured e2 ]) expsTup) in
+      Item (SOME tra) "Mat" [pres_to_structured exp; expsTup'])
+  /\
+  (pres_to_structured (Let tra varN exp1 exp2) =
+    let varN' = option_string_to_structured varN in
+      Item (SOME tra) "Let" [varN'; pres_to_structured exp1; pres_to_structured exp2])
+  /\
+  (pres_to_structured (Letrec tra varexpTup exp) =
+    let varexpTup' = List (MAP (\ (v1, v2, e) . Tuple [
+      string_to_structured v1;
+      string_to_structured v2;
+      pres_to_structured e
     ]) varexpTup) in
-      new_obj "Letrec" [("tra", trace_to_json tra); ("varsexp", varexpTup'); ("exp", pres_to_json exp)])
+      Item (SOME tra) "Letrec" [varexpTup'; pres_to_structured exp])
   /\
-  (pres_to_json _ = Null)`
-  cheat;
+  (pres_to_structured _ = Item NONE "\"Unknown constructor\"" [])
```

Comment:
Same as above, with `op_to_structured` -- I think it is better if we get an error on the HOL side of things, rather than this string appearing deep down in the JSON somewhere.